### ISSUE TITLE
Blur widget for hardcoded subtitles

### ISSUE THREAD

**Issue by cristian428** [killergerbah/asbplayer#444]
I understand there's a planned update to add a blur for subtitle tracks and I think that's great but something I've thought would be very useful before is having a dedicated blur widget to deal with hardcoded subs. Especially with Chinese language content, there's often hardcoded subs. In today's case I'm watching a show with subs in both ZH and EN. English subtitles are not wanted so having the option to hide them would be great. On windows PC there's a cool program that lets you put a blur bar over windows and I do use that on PC when needed but there's no equivalent of that for phones and tablets that I know of. At least for me, hardcoded subs can be very distracting.

![Screenshot_20240621_235131_YouTube](https://github.com/killergerbah/asbplayer/assets/36088384/41d49a41-aeb7-4577-a8e0-0c7dafa28db7)

**Comment by artjomsR** [comment]
I wonder if this can be achieved to a satisfactory degree using the existing blur settings.

- load your subtitles twice, one displayed at the top, other - at the bottom
- Top subtitles are your actual subtitles that you want. Configure as needed
- Bottom subtitles are your blur widget. Enable the blur setting for them, increase the font size (increasing the "blur widget's" size as a result) and adjust the spacing at the bottom as needed

If you find that hardcoded subs are still occasionally visible, then it must be that the timings on hard and soft subs don't match. So you could create a sub file which would display a long subtitle throughout the whole video, e.g.
``` blur.srt
WEBVTT

1
00:00:00.000 --> 99:99:99.000
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

**Comment by artjomsR** [comment]
Duplicate of https://github.com/killergerbah/asbplayer/issues/444

**Comment by shiki-tm** [comment]
Huh? 
I only ever talked about hardcoded subs. 
So there's no subs to load.

**Comment by artjomsR** [comment]
I get that. Read again - you'd take the subtitles that you already have and load them twice (you can copy/ paste them and load both at the same time or load them one by one, either way effect is the same).

The top sub serves its original purpose to show the subs you wanted. And by adding blur styling to the bottom subs, you can effectively create a blur overlay from your original subs

**Comment by killergerbah** [comment]
Yeah it could be done manually. Maybe use "______" for the text if you want to mouse-over and unblur to see the harcoded subs behind. Obviously would still be nice if asbplayer provided a one button option to do this.

**Comment by artjomsR** [comment]
The PR was merged but this only applies to the app, not the extension, or is there a way to use it in the extension as well?

**Comment by killergerbah** [comment]
Yes, it could be added. It's a UI challenge, just need to figure out a way to fit more buttons.
